agent/internal/container/container.go

```diff
@@ -218,9 +218,14 @@ func (c *Container) run(parent context.Context) (err error) {
 			if err != nil {
 				c.log.Trace("ensuring cleanup of container (canceled prior to the monitoring loop)")
 				if remove {
-					// TODO this context could be canceled between CreateContainer and this.
-					if rErr := c.docker.RemoveContainer(ctx, dockerID, true); rErr != nil {
-						c.log.WithError(rErr).Debug("couldn't cleanup container")
+					if ctx.Err() == context.Canceled {
```

Comment:
Sounds good! I had been unsure if it was safe to do so in case the desired outcome was to keep the container going detached. I'll get that swapped over 😄